modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDatabase.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.ingest.geoip;
+
+import com.maxmind.geoip2.model.AsnResponse;
+import com.maxmind.geoip2.model.CityResponse;
+import com.maxmind.geoip2.model.CountryResponse;
+
+import org.elasticsearch.core.Nullable;
+
+import java.io.IOException;
+import java.net.InetAddress;
+
+/**
+ * Provides a uniform interface for interacting with various GeoIP databases.
+ */
+public interface GeoIpDatabase {
+
+    /**
+     * @return the database type as it is detailed in the database file metadata
+     * @throws IOException if the database file could not be read or the data could not be accessed
+     */
+    String getDatabaseType() throws IOException;
```

Comment:
I'm not really familiar with the type. Is there a way to make it an enum (city, country, etc)? Or is it really open-ended?

Reply:
This is specifically the type that is listed in the Maxmind database file metadata. From the looks of things it identifies the original filename in the cases I tested. There is also a database type enum that exists in the Maxmind libraries but I don't know how difficult it is to lazily load that type enum from the database file.

Reply:
A _vague_ memory from digging in the maxmind code ~8 years ago gives me the spidey-sense that the database name encodes _all_ of the types of responses you can get out of it, including what is also available from commercially-licensed variants? It is a very vague memory at this point.